### ISSUE TITLE
Bug: Unable to run nighthawk due to envoy warning

### ISSUE THREAD

**Issue by kushthedude** [envoyproxy/nighthawk#597]
*Title*: *One line description*

*Description*:
After executing nighthawk_client, there are several envoy errors which stop the nighthawk execution for further processing.

*Reproduction steps*:

Run 
```sh
docker run envoyproxy/nighthawk-dev:latest nighthawk_client --rps 3 --concurrency 5 --duration 30 https://github.com --output-format experimental_fortio_pedantic
```

*Logs*:

```sh
[2020-12-15 13:27:24.096][1][warning][misc] [external/envoy/source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type nighthawk.client.CommandLineOptions Using deprecated option 'nighthawk.client.CommandLineOptions.allow_envoy_deprecated_v2_api' from file options.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.
[13:27:24.130053][1][I] Starting 5 threads / event loops. Time limit: 30 seconds.
[13:27:24.130598][1][I] Global targets: 500 connections and 15 calls per second.
[13:27:24.131281][1][I]    (Per-worker targets: 100 connections and 3 calls per second)
[13:27:24.133120][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[13:27:24.184795][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[13:27:24.206339][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[13:27:24.229789][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[13:27:24.253361][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[13:27:49.813616][12][E] Exiting due to failing termination predicate
[13:27:49.815278][12][I] Stopping after 24885 ms. Initiated: 75 / Completed: 75. (Completion rate was 3.0138282880575287 per second.)
[13:27:49.843730][13][E] Exiting due to failing termination predicate
[13:27:49.844784][13][I] Stopping after 24848 ms. Initiated: 75 / Completed: 75. (Completion rate was 3.018254281162298 per second.)
[13:27:49.918637][14][E] Exiting due to failing termination predicate
[13:27:49.919979][14][I] Stopping after 24857 ms. Initiated: 75 / Completed: 75. (Completion rate was 3.017253500345958 per second.)
[13:27:49.981517][15][E] Exiting due to failing termination predicate
[13:27:49.981998][15][I] Stopping after 24853 ms. Initiated: 75 / Completed: 75. (Completion rate was 3.0177154381252054 per second.)
[13:27:50.076447][11][E] Exiting due to failing termination predicate
[13:27:50.076855][11][I] Stopping after 25214 ms. Initiated: 76 / Completed: 76. (Completion rate was 3.014097449339863 per second.)
```

**Comment by kushthedude** [comment]
Related Issue https://github.com/envoyproxy/envoy/issues/13504

**Comment by oschaaf** [comment]
Trying that docker command, it actually works for me. The reason the command fails in your case is indicated by:

```
[13:27:49.843730][13][E] Exiting due to failing termination predicate
```

It would be helpful to post (a link to) the output of the docker command when also passing `-v trace` to nighthawk. I suspect that somehow the client is not able to connect to https://github.com/.

**Comment by kushthedude** [comment]
The error occur on using the nighthawk binary command too.

**Comment by kushthedude** [comment]
> Trying that docker command, it actually works for me. The reason the command fails in your case is indicated by:
> 
> ```
> [13:27:49.843730][13][E] Exiting due to failing termination predicate
> ```
> 
> It would be helpful to post (a link to) the output of the docker command when also passing `-v trace` to nighthawk. I suspect that somehow the client is not able to connect to https://github.com/.

Tried other URLs same issue 
```sh
[2020-12-15 20:46:48.256][1][warning][misc] [external/envoy/source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type nighthawk.client.CommandLineOptions Using deprecated option 'nighthawk.client.CommandLineOptions.allow_envoy_deprecated_v2_api' from file options.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.
[20:46:49.306600][1][I] Starting 5 threads / event loops. Time limit: 30 seconds.
[20:46:49.307164][1][I] Global targets: 500 connections and 15 calls per second.
[20:46:49.307890][1][I]    (Per-worker targets: 100 connections and 3 calls per second)
[20:46:49.310350][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[20:46:49.350190][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[20:46:49.393728][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[20:46:49.432127][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[20:46:49.454021][1][W] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
[20:46:50.900074][11][E] Exiting due to failing termination predicate
[20:46:50.901626][11][I] Stopping after 840 ms. Initiated: 3 / Completed: 3. (Completion rate was 3.5676655337346563 per second.)
[20:46:50.953271][12][E] Exiting due to failing termination predicate
[20:46:50.954256][12][I] Stopping after 827 ms. Initiated: 2 / Completed: 2. (Completion rate was 2.417073238527664 per second.)
[20:46:51.020913][13][E] Exiting due to failing termination predicate
[20:46:51.021522][13][I] Stopping after 828 ms. Initiated: 2 / Completed: 2. (Completion rate was 2.4144004500442438 per second.)
[20:46:51.090515][14][E] Exiting due to failing termination predicate
[20:46:51.091618][14][I] Stopping after 831 ms. Initiated: 2 / Completed: 2. (Completion rate was 2.4061047690199575 per second.)
[20:46:51.150216][15][E] Exiting due to failing termination predicate
[20:46:51.152897][15][I] Stopping after 824 ms. Initiated: 2 / Completed: 2. (Completion rate was 2.426274613039528 per second.)
```

Here is the log dump of `-v trace`
[nighthawk_output.txt](https://github.com/envoyproxy/nighthawk/files/5698582/nighthawk_output.txt)

**Comment by oschaaf** [comment]
In the logs the target got moved from 'https://github.com/' to `https://gitter.com/`, right? 
I think I noticed gitter.com redirecting to another host before I attempted to reproduce; however that shouldn't be a problem. But now that I tried to repro,  I can't check again: the host is blocking access for me. So I can reproduce what you observe there: I theorize their firewall hates it when it gets spammed with synthetic traffic? :-)
Having said all that, github.com still seems to be working fine for me in a cautious test, but I wouldn't recommend flooding that host with synthetic traffic for obvious reasons.
